src/main/scala/diplomacy/package.scala

```diff
@@ -7,6 +7,168 @@ import chisel3.Data
+  *      |  |                           :  ↓  +                             |      |
+  *      |  |  Upwards-flowing output   :  ↓  + Downwards-Flowing output    |      |
+  *      |  |    parameters of type UO  :  ↓  + parameters of type DO       |      |
+  *      |  |---------------------------:--↓--+-----------------------------|      |          
+  *      |                              :  ↓  +                                    |
+  *      |                              :  ↓  +                                    |
+  *      |------------------------------:--↓--+------------------------------------|
+  *                                     :  ↓  v                               
+  *                                        ↓ outward edge of (parameter) type EO
+  *                                        ↓ created from parameters of type UO and DO
+  *                                        ↓ will result in a Bundle of type BO
+  *                                        ↓
+  *                                        ↓                          Downwards (aka Sink)
+  *
+  * }}}
+  * 
+  * == Handles ==
+  *
+  * Two Diplomatic nodes can be bound together using the `:=` operator or one of
+  * its sibling operators. Binding is asymmetric, and the binding operation will
+  * connect the outer side of one node to the inner side of the other.
+  *
+  * For example, the expression `a := b` will connect the outer side of `b` to the
+  * inner side of `a`.
+  *
+  * We would like the `:=` operator to have additional properties that make it
+  * intuitive to use:
+  *
+  * 1. It should be chainable, so that `a := b := c` will have the intuitive effect
+  *    of binding c to b and b to a. This requires that the return type of `:=` be the
+  *    same as its arguments, because the result of one `:=` operation must be
+  *    valid as an argument to other `:=` operations.
+  *
+  * 2. It should be associative, so that `(a := b) := c` is equivalent to `a := (b := c).`
+  *    This means that the order in which the bind operations execute does
+  *    not matter, even if split across multiple files.
+  *
+  * 3. `a := b` should only be allowed if and only if `b` allows outward edges and `a`
+  *    allows inward edges. This should be preserved even when chaining
+  *    operations, and it should ideally be enforced at compile time.
+  *
+  * [[Handle]]s are a way of satisfying all of these properties. A Handle represents
+  * the aggregation of a chain of Nodes, and it preserves information about
+  * the connectability of the innermost and the outermost sides of the chain.
+  *
+  * If `b` supports inward edges, then `a := b` returns a [[Handle]] that supports inward
+  * edges that go into `b`. If `a` supports outward edges, then `a := b` returns a
+  * [[Handle]] that supports outward edges coming out of `a`.
+  *
+  *  ==Node Terms==
+  *
+  *  These are some conventionally used terms for diplomatic Nodes:
```

Comment:
`...which describe different common subtypes that certain protocol implementation might utilize:`